scripts/yul_coverage.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -e
+
+ROOT_DIR="$(dirname "$0")"/..
+
+FAILED=()
+SUCCESS=()
+test_file()
+{
+  local SOL_FILE
+  local OUTPUT
+
+  SOL_FILE=${1}
+  OUTPUT=$(solc --ir "${SOL_FILE}" 2>&1 || true)
+  if [[ ${OUTPUT} == *"UnimplementedFeatureError"* ]]; then
```

Comment:
Would it make sense to test the executable return value instead?

Reply:
I agree, but there two things in the output that probably cause it to exit with non-zero status even if `UnimplementedFeatureError` is not present:
```
Internal compiler error during compilation: 
```
```
Warning: This is a pre-release compiler version, please do not use it in production. Warning: Source file does not specify required compiler version! -->
```
I am not sure if treating them as success is intended. For the warning it probably is.

Reply:
Makes sense. I don't know about the second, but the first shouldn't happen anyway (otherwise it's an actual bug)

Reply:
On my machine I have a lot of errors like these if I add a line to echo `$OUTPUT` to stdout:
```
Internal compiler error during compilation: /solidity/libsolidity/ast/Types.cpp(2920): Throw in function virtual bool solidity::frontend::FunctionType::leftAligned() const Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError> std::exception::what: Alignment property of non-exportable function type requested. [solidity::util::tag_comment*] = Alignment property of non-exportable function type requested.
```
```
Internal compiler error during compilation: /solidity/libsolidity/codegen/ir/IRGeneratorForStatements.cpp(956): Throw in function virtual void solidity::frontend::IRGeneratorForStatements::endVisit(const solidity::frontend::MemberAccess&) Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError> std::exception::what: Member access to unknown type. [solidity::util::tag_comment*] = Member access to unknown type.
```
but there are no errors when I run full test suite. Does this mean that there are files in `syntaxTests/` and `semanticTests/` that the test suite is not running on?

Reply:
Oh wow, `Internal compiler error` should really not happen, will check this.